### PR TITLE
using VRF to get random number & edit some functions

### PR DESCRIPTION
**Pull request by yuchen3890**
- add variable upgradeFee, topLevel
- get random number from VRF
- check require
- edit some functions

### CODE REVIEW

File: contracts/Enhancement.sol [hunk 1]
```diff
@@ -4,55 +4,71 @@ pragma solidity ^0.8.0;
...
+    uint8 public topLevel;
     bool public paused = false;
     bool public useDynamicUri = true;
     string public staticUri;
+    // for getRandomNumber
+    bytes32 public keyHash;
+    uint256 public fee;
+    uint256 public randomResult;
 
     mapping(uint256 => uint8) tokenIdToLevel;
```
Comment (morris1961) [inline contracts/Enhancement.sol]: Please add public for me

File: contracts/Enhancement.sol [hunk 2]
```diff
@@ -152,13 +162,27 @@ contract Enhancement is ERC721Enumerable, PullPayment, Ownable {
         return false;
     }
 
+    // Get random number from ChainLink node
+    function getRandomNumber() public returns (bytes32 requestId) {
```
Comment (morris1961) [inline contracts/Enhancement.sol]: The function should be internal

File: contracts/Enhancement.sol [hunk 3]
```diff
@@ -152,13 +162,27 @@ contract Enhancement is ERC721Enumerable, PullPayment, Ownable {
...
+        return requestRandomness(keyHash, fee);
+    }
+
+    // Verify random number through VRF coordinator
+    function fulfillRandomness(bytes32 requestId, uint256 randomness) internal override {
```
Comment (morris1961) [inline contracts/Enhancement.sol:177]: Don't know what the function for

File: scripts/deploy.js [hunk 4]
```diff
@@ -11,10 +11,12 @@ task("deploy", "Deploys the NFT.sol contract")
 	.addParam("name", "The name of the contract", "Enhancement")
 	.addParam("symbol", "The symbol of the contract", "EHM")
 	.addParam("price", "The price of the NFT", "100000000000000")
+	.addParam("upgradeFee", "The fee to upgrade your NFT", "0")
+	.addParam("topLevel", "The highest level")
```
Comment (morris1961) [inline scripts/deploy.js]: topLevel should default be 10

**Review by morris1961 — CHANGES_REQUESTED** [review]

File: contracts/Enhancement.sol (hunk 2 above)
Comment (yuchen3890) [reply contracts/Enhancement.sol]: Ok. Changed.

File: scripts/deploy.js (hunk 4 above)
Comment (yuchen3890) [reply scripts/deploy.js]: Sorry. It's done now.

File: contracts/Enhancement.sol (hunk 3 above)
Comment (yuchen3890) [reply contracts/Enhancement.sol:177]: I'll explain u later.

**Review by morris1961 — APPROVED** [review]
No problem please merge

**Pull request merged**